nixos/modules/services/mail/roundcube.nix

```diff
@@ -142,26 +148,23 @@ in
       ${cfg.extraConfig}
     '';
 
-    services.nginx = {
+    services.nginx = lib.mkIf cfg.configureNginx {
       enable = true;
       virtualHosts = {
         ${cfg.hostName} = {
-          forceSSL = mkDefault true;
-          enableACME = mkDefault true;
+          root = cfg.package;
           locations."/" = {
-            root = cfg.package;
             index = "index.php";
             extraConfig = ''
-              location ~* \.php(/|$) {
```

Comment:
What we should do here though is to return a 404 on stuff that shouldn't be served, i.e. `bin`/`COPYING`/etc.. See also https://wiki.archlinux.org/title/Roundcube.

Reply:
Can simply change the root directory to `${cfg.package}/public_html`. But this will lead to errors in the operation of the site.
```
Dec 24 09:46:21 test roundcube[92647]: Stack trace:
Dec 24 09:46:21 test roundcube[92647]: #0 /nix/store/v4y7p4y4yc1r4sa9vv21xhsmdsj8h984-roundcube-1.6.5/plugins/zipdownload/zipdownload.php(58): zipdownload->download_menu()
Dec 24 09:46:21 test roundcube[92647]: #1 /nix/store/v4y7p4y4yc1r4sa9vv21xhsmdsj8h984-roundcube-1.6.5/program/lib/Roundcube/rcube_plugin_api.php(105): zipdownload->init()
Dec 24 09:46:21 test roundcube[92647]: #2 /nix/store/v4y7p4y4yc1r4sa9vv21xhsmdsj8h984-roundcube-1.6.5/program/include/rcmail.php(160): rcube_plugin_api->init()
Dec 24 09:46:21 test roundcube[92647]: #3 /nix/store/v4y7p4y4yc1r4sa9vv21xhsmdsj8h984-roundcube-1.6.5/program/include/rcmail.php(93): rcmail->startup()
Dec 24 09:46:21 test roundcube[92647]: #4 /nix/store/v4y7p4y4yc1r4sa9vv21xhsmdsj8h984-roundcube-1.6.5/index.php(43): rcmail::get_instance()
Dec 24 09:46:21 test roundcube[92647]: #5 /nix/store/v4y7p4y4yc1r4sa9vv21xhsmdsj8h984-roundcube-1.6.5/public_html/index.php(26): include('...')
Dec 24 09:46:21 test roundcube[92647]: #6 {main}
```
When adding plugins and using symbolic links, the result is that the roundcube package without plugins is used.
To solve this problem, need to build the rouncube with plugins package without using symbolic links.

Reply:
> What we should do here though is to return a 404 on stuff that shouldn't be served, i.e. `bin`/`COPYING`/etc.. See also https://wiki.archlinux.org/title/Roundcube.

Update.